src/extensions/relations/Relation.js

```diff
@@ -0,0 +1,1165 @@
+/**
```

Comment:
We're using this class on editor now as well, so decided to split it and move from `js` folder. Saved history of changes, but gh diff is broken anyway (

Use sth like https://www.diffchecker.com/ to compare difference if needed 👍